fly/commands/internal/flaghelpers/job_flag.go

```diff
@@ -12,19 +12,37 @@ import (
 )
 
 type JobFlag struct {
-	PipelineName string
-	JobName      string
+	PipelineRef atc.PipelineRef
+	JobName     string
 }
 
-func (job *JobFlag) UnmarshalFlag(value string) error {
-	vs := strings.SplitN(value, "/", -1)
+func (flag *JobFlag) UnmarshalFlag(value string) error {
+	flag.PipelineRef = atc.PipelineRef{}
 
-	if len(vs) != 2 {
+	jobNameIdx := strings.LastIndex(value, "/")
+	if jobNameIdx == -1 {
 		return errors.New("argument format should be <pipeline>/<job>")
 	}
 
-	job.PipelineName = vs[0]
-	job.JobName = vs[1]
+	flag.JobName = value[jobNameIdx+1:]
+	if warnings := atc.ValidateIdentifier(flag.JobName, "job"); warnings != nil {
+		return errors.New("argument format should be <pipeline>/<key:value>/<job>")
+	}
```

Comment:
Hmm I suspect we don't want to be strict about the job/resource identifier yet

Reply:
Oh haha, think I just haven't read far enough into the commits, as it looks like this already may have changed